--- Crypt-OpenPGP-1.01/Makefile.PL.orig	Sat Aug  3 22:49:15 2002
+++ Crypt-OpenPGP-1.01/Makefile.PL	Sat Aug  3 22:49:29 2002
@@ -156,7 +156,7 @@
 
         while (my($k, $v) = each %todo) {
             if (prompt(sprintf("%s%s not installed/out of date. Install (y/n)?",
-              $k, $v ? " (ver. $v)" : ""), "y") =~ /^y/) {
+              $k, $v ? " (ver. $v)" : ""), "n") =~ /^y/) {
                 require CPAN; CPAN::Shell->install($k);
                 delete $prereq{$k};
                 chdir $cwd or die "Can't chdir back to $cwd: $!";
